auth/src/main/java/org/aerogear/mobile/security/SecurityCheckMetricPublisher.java

```diff
@@ -32,6 +32,7 @@ public synchronized void onExecuted(SecurityCheckResult result) {
 
     @Override
     public synchronized void onComplete() {
```

Comment:
This is how it is called:
`
        final AtomicInteger count = new AtomicInteger(checks.size());
        for (final SecurityCheck check : checks) {
            res.put(check.getId(), (executorService.submit(() -> {
                final SecurityCheckResult result = check.test(getContext());

                final int remaining = count.decrementAndGet();

                for (SecurityCheckExecutorListener listener : listeners) {
                    listener.onExecuted(result);

                    if (remaining <= 0) {
                        listener.onComplete();
                    }
                }

                return result;
            })));
        }
`


`onExecuted` above is sync because it could be called from any one of the checks and it may result in `ConcurrentModificationException` because of the list in there. But can't tell the same thing for `onComplete`. IMO, it can be non-sync.
However, I can see that there might be a case where `onComplete` is called more than once. WDYT?

BTW, I haven't changed this behavior, so let's get opinion from @danielpassos